[Profiler] add config option to remove 'Call stack' field from trace file

slgong-fb opened the pull request:
Summary: `Call stack` field increases trace file size exponentially for Python stack tracing (need to be deprecated carefully). Added a config option to avoid this increase.

Test Plan:
`experimental_config=_ExperimentalConfig(no_callstack_trace=True),` will remove the field.
+ CI tests

Differential Revision: D39489828

robieta commented on torch/_C/_profiler.pyi:
```diff
@@ -35,6 +35,7 @@ class _ExperimentalConfig:
         self,
         profiler_metrics: List[str] = ...,
         profiler_measure_per_kernel: bool = ...,
+        no_callstack_trace: bool = ...,
```
I really like the idea of putting this flag in ExperimentalConfig. I think we should make it more generic. Maybe `new_trace_format`? (@aaronenyeshi, @chaekit feel free to bikeshed.)

slgong-fb replied:
will mark resolved unless we are trying `new_trace_format` in this PR.

robieta replied:
I was mostly thinking just because that way we wouldn't have to update the name every time we want to improve the trace format. The code itself doesn't have to change. (Maybe `updated_format`?)

slgong-fb replied:
Good point. I chose `verbose` (more general) for this PR changes instead of `no_callstack_trace`, with an opposite default value and comment. I guess we would update trace format in the future but not only just reducing it but improve in a different way?

robieta replied:
Ohh, I like `verbose`.

robieta commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -85,8 +86,12 @@ struct MetadataBase {
+      const std::string& key,
+      const std::string& value,
+      const bool no_callstack_trace = false) {
+    if (!no_callstack_trace && kineto_activity_ && !value.empty() &&
```
`addMetadata` is a generic method. Config based omission should be handled by the caller.

slgong-fb replied:
okay

robieta commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -102,12 +107,16 @@ struct MetadataBase {
     const auto module_hierarchy = kineto_event.moduleHierarchy();
     addMetadata("Module Hierarchy", stacksToStr(module_hierarchy.vec(), "."));
-    addMetadata("Call stack", stacksToStr(kineto_event.stack().vec(), ";"));
+    addMetadata(
```
This would also drop call stack from JIT, which we want to preserve. (Since the end goal is to flip a switch.)

slgong-fb replied:
I changed to drop call stack info is KinetoEvent constructor, where `python_stack_` is built, and changes are working fine in my local test.

robieta reviewed (changes requested):
High level thought: perhaps the simplest thing to do would be to change the signature of `KinetoEvent::stack()` to `KinetoEvent::stack(bool jit_only=false)`. Then you could just pass the config value to `stack` when generating metadata.

robieta commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -618,16 +620,19 @@ std::unique_ptr<ProfilerResult> disableProfiler() {
-    parent->visit_if_base<PyExtraFieldsBase>(
-        [&](const auto& i) { python_stack_.push_back(parent->name()); });
-    parent = parent->parent_.lock();
+  if (!no_callstack_trace) {
```
not no callstack trace is kind of hard to parse.

robieta reviewed (approved):
I had a couple closing comments, but nothing major. So if you want to just submit and follow up later you're welcome to.

slgong-fb committed:
[Profiler] add config option to remove 'Call stack' field from trace file (#84982)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/84982

`Call stack` field increases trace file size exponentially when python call stacks are deep (need to be deprecated carefully). Added a config option to avoid this increase.

Test Plan:
`experimental_config=_ExperimentalConfig(verbose=False),` will remove the field in the trace file.
+ CI tests

Differential Revision: D39489828

fbshipit-source-id: 5b26e5c2aed93b6c38b9466ee7ac23a56e16f8a8

slgong-fb commented:
@pytorchbot merge -g
